packages/@react-aria/interactions/src/PressResponder.tsx

```diff
@@ -38,12 +38,14 @@ export const PressResponder = React.forwardRef(({children, ...props}: PressRespo
   useSyncRef(prevContext, ref);
 
   useEffect(() => {
-    if (!isRegistered.current) {
-      console.warn(
+    if (process.env.NODE_ENV !== 'production') {
```

Comment:
This one is a bit odd. Would be nice to drop the entire useEffect. We'd need to disable the rules of hooks lint check, because we wouldn't actually be breaking the rule since it'd be consistent within an environment.